src/components/Composer/index.js

```diff
@@ -86,387 +85,336 @@ const propTypes = {
+        // when navigating away to different report
+        const unsubscribeFocus = navigation.addListener('focus', () => document.addEventListener('paste', handlePaste));
+        const unsubscribeBlur = navigation.addListener('blur', () => document.removeEventListener('paste', handlePaste));
+
+        if (_.isFunction(forwardedRef)) {
+            forwardedRef(textInput.current);
+        }
 
-        this.updateNumberOfLines();
-    }
-
-    /**
-     * Check the current scrollHeight of the textarea (minus any padding) and
-     * divide by line height to get the total number of rows for the textarea.
-     */
-    updateNumberOfLines() {
-        // Hide the composer expand button so we can get an accurate reading of
-        // the height of the text input
-        this.props.setIsFullComposerAvailable(false);
-
-        // We have to reset the rows back to the minimum before updating so that the scrollHeight is not
-        // affected by the previous row setting. If we don't, rows will be added but not removed on backspace/delete.
-        this.setState({numberOfLines: 1}, () => {
-            const computedStyle = window.getComputedStyle(this.textInput);
-            const lineHeight = parseInt(computedStyle.lineHeight, 10) || 20;
-            const paddingTopAndBottom = parseInt(computedStyle.paddingBottom, 10) + parseInt(computedStyle.paddingTop, 10);
-            const computedNumberOfLines = ComposerUtils.getNumberOfLines(this.props.maxLines, lineHeight, paddingTopAndBottom, this.textInput.scrollHeight);
-            const numberOfLines = computedNumberOfLines === 0 ? this.props.numberOfLines : computedNumberOfLines;
-            updateIsFullComposerAvailable(this.props, numberOfLines);
-            this.setState({
-                numberOfLines,
-                width: computedStyle.width,
-            });
-            this.props.onNumberOfLinesChange(numberOfLines);
-        });
-    }
+        if (textInput.current) {
+            textInput.current.addEventListener('paste', handlePaste);
+            textInput.current.addEventListener('wheel', handleWheel);
+        }
 
-    render() {
-        const propStyles = StyleSheet.flatten(this.props.style);
-        propStyles.outline = 'none';
-        const propsWithoutStyles = _.omit(this.props, 'style');
-
-        // This code creates a hidden text component that helps track the caret position in the visible input.
-        const renderElementForCaretPosition = (
-            <View
-                style={{
-                    position: 'absolute',
-                    bottom: -2000,
-                    zIndex: -1,
-                    opacity: 0,
-                }}
+        return () => {
+            unsubscribeFocus();
+            unsubscribeBlur();
+            document.removeEventListener('paste', handlePaste);
+            // eslint-disable-next-line es/no-optional-chaining
+            textInput.current?.removeEventListener('wheel', handleWheel);
```

Comment:
Is optional chaining needed here at all? 

```suggestion
            if(!textInput.current) {
                return;
            }
            textInput.current.removeEventListener('wheel', handleWheel);
```

Reply:
👍🏻